File: gofiles/controller/AuthController.go
package controller

import (
	"fmt"
	"net/http"

	"github.com/gdpl2112/files-server/model"
	"github.com/gdpl2112/files-server/service"
	"github.com/gin-gonic/gin"
)

// AuthController 认证控制器
type AuthController struct {
	AuthService   *service.AuthService
	AuthServerUrl string
	AppID         string
	RedirectUri   string
}

// NewAuthController 创建认证控制器实例
func NewAuthController(authService *service.AuthService, authServerUrl, appID, redirectUri string) *AuthController {
	return &AuthController{
		AuthService:   authService,
		AuthServerUrl: authServerUrl,
		AppID:         appID,
		RedirectUri:   redirectUri,
	}
}

// Register 注册路由
func (c *AuthController) Register(router *gin.Engine) {
	authGroup := router.Group("/auth")
	{
		authGroup.GET("/login", c.Login)
		authGroup.GET("/callback", c.Callback)
		authGroup.GET("/user", c.GetCurrentUser)
		authGroup.POST("/logout", c.Logout)
	}
}

// Login 重定向到授权服务器登录
func (c *AuthController) Login(ctx *gin.Context) {
	authorizeUrl := fmt.Sprintf("%s/authc?app_id=%s&redirect_uri=%s", c.AuthServerUrl, c.AppID, c.RedirectUri)
	ctx.Redirect(http.StatusFound, authorizeUrl)
}

// Callback 处理授权服务器回调
func (c *AuthController) Callback(ctx *gin.Context) {
	code := ctx.Query("code")
	if code == "" {
		ctx.JSON(http.StatusBadRequest, gin.H{"message": "缺少授权码"})
		return
	}

	user := c.AuthService.HandleAuthCallback(code)
	if user == nil {
		ctx.JSON(http.StatusUnauthorized, gin.H{"message": "登录失败"})
		return
	}

	// 将用户信息存储到会话中
	ctx.Set("user", user)
	ctx.Redirect(http.StatusFound, "/")
}

// GetCurrentUser 获取当前用户信息
func (c *AuthController) GetCurrentUser(ctx *gin.Context) {
	user, exists := ctx.Get("user")
	if !exists {
		ctx.JSON(http.StatusUnauthorized, gin.H{"message": "未登录"})
		return
	}

	currentUser, ok := user.(*model.User)
	if !ok {
		ctx.JSON(http.StatusInternalServerError, gin.H{"message": "用户信息类型错误"})
		return
	}

	ctx.JSON(http.StatusOK, gin.H{
		"userId":       currentUser.UserID,
		"username":     currentUser.Username,
		"accessToken":  currentUser.AccessToken,
		"storageLimit": currentUser.StorageLimit,
		"usedSpace":    currentUser.UsedStorage,
	})
}

// Logout 用户登出
func (c *AuthController) Logout(ctx *gin.Context) {
	// 清除会话中的用户信息
	ctx.Set("user", nil)
	ctx.JSON(http.StatusOK, gin.H{"message": "已登出"})
}

File: src/main/java/io/github/gdpl2112/AuthController.java
package io.github.gdpl2112;

import io.github.gdpl2112.model.User;
import io.github.gdpl2112.service.AuthService;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.servlet.http.HttpSession;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/auth")
public class AuthController {
    
    @Autowired
    private AuthService authService;
    
    @Value("${server.auth-server}")
    private String authServerUrl;
    
    @Value("${auth.app-id}")
    private String appId;
    
    @Value("${auth.redirect-uri}")
    private String redirectUri;
    
    /**
     * 重定向到授权服务器进行登录
     * @return 重定向URL
     */
    @GetMapping("/login")
    public ResponseEntity<String> login(HttpServletResponse response) throws IOException {
        String authorizeUrl = String.format("%s/authc?app_id=%s&redirect_uri=%s", authServerUrl, appId, redirectUri);
        response.sendRedirect(authorizeUrl);
        return ResponseEntity.status(302).body(authorizeUrl);
    }
    
    /**
     * 处理授权服务器的回调
     * @param code 授权码
     * @param session HTTP会话
     * @return 登录结果
     */
    @GetMapping("/callback")
    public ResponseEntity<String> callback(@RequestParam("code") String code, HttpSession session, HttpServletResponse response) throws IOException {
        User user = authService.handleAuthCallback(code);
        if (user != null) {
            session.setAttribute("user", user);
            response.sendRedirect("/");
            return ResponseEntity.ok("登录成功，欢迎 " + user.getUsername());
        } else {
            return ResponseEntity.status(401).body("登录失败");
        }
    }
    
    /**
     * 获取当前用户信息
     * @param session HTTP会话
     * @return 当前用户信息
     */
    @GetMapping("/user")
    public ResponseEntity<Map<String, Object>> getCurrentUser(HttpSession session) {
        User user = (User) session.getAttribute("user");
        if (user != null) {
            Map<String, Object> userInfo = new HashMap<>();
            userInfo.put("userId", user.getUserId());
            userInfo.put("username", user.getUsername());
            userInfo.put("accessToken", user.getAccessToken());
            userInfo.put("storageLimit", user.getStorageLimit());
            userInfo.put("usedSpace", user.getUsedStorage());
            return ResponseEntity.ok(userInfo);
        } else {
            return ResponseEntity.status(401).build();
        }
    }
    
    /**
     * 用户登出
     * @param session HTTP会话
     * @return 登出结果
     */
    @PostMapping("/logout")
    public ResponseEntity<String> logout(HttpSession session) {
        session.invalidate();
        return ResponseEntity.ok("已登出");
    }
}